### ISSUE TITLE
Clang assertion fails generating IR for atomic builtins

### ISSUE THREAD

**Issue by davidchisnall** [CTSRD-CHERI/clang#118]
Test case:

```
void fail(void)
{
 _Atomic(_Bool) obj;
 __c11_atomic_exchange(&obj, 1, 5);
}
```

Generates an assertion failure because of an invalid bitcast from a pointer to an integer.

Compile with `-mabi=sandbox` to trigger the assertion failure.

**Comment by davidchisnall** [comment]
Bug appears to be `CGAtomic.cpp:1065` - cast is ignoring address space.

**Comment by davidchisnall** [comment]
Fixed in 335556b8170e39e6d720dcbda1d43a4533aee048
